File: backend/app/schemas/export.py
from uuid import UUID

from pydantic import BaseModel, Field


class ApplyEditsRequest(BaseModel):
    statement_id: UUID
    session_id: str | None = Field(
        default=None,
        description="Active or committed edit session. Uses committed metadata if omitted.",
    )
    validate_visual: bool = True


class VisualValidationSchema(BaseModel):
    text_match_ratio: float
    bbox_overlap_ratio: float
    regions_checked: int
    issues: list[str]
    passed: bool


class ApplyEditsResponse(BaseModel):
    statement_id: UUID
    status: str
    download_url: str
    original_preview_url: str
    edited_preview_url: str
    replacements_applied: int
    replacements_failed: int
    validation: VisualValidationSchema
    warnings: list[str] = Field(default_factory=list)

File: frontend/src/types/export.ts
export interface ApplyEditsRequest {
  statement_id: string
  session_id?: string | null
  validate_visual?: boolean
}

export interface VisualValidation {
  text_match_ratio: number
  bbox_overlap_ratio: number
  regions_checked: number
  issues: string[]
  passed: boolean
}

export interface ApplyEditsResponse {
  statement_id: string
  status: string
  download_url: string
  original_preview_url: string
  edited_preview_url: string
  replacements_applied: number
  replacements_failed: number
  validation: VisualValidation
  warnings: string[]
}
